editoast/src/views/infra/mod.rs

```diff
@@ -543,15 +541,25 @@ async fn get_switch_types(
     })
     .await?;
 
-    let infra = InfraCache::get_or_load(conn, &infra_caches, &infra).await?;
-    Ok(Json(
-        infra
-            .switch_types()
-            .values()
-            .map(ObjectCache::unwrap_switch_type)
-            .cloned()
-            .collect(),
-    ))
+    let selection_settings =
+        SelectionSettings::new().filter(move || SwitchTypeModel::INFRA_ID.eq(infra.id));
+    let switch_types_model = SwitchTypeModel::list(conn, selection_settings)
+        .await
+        .expect("Failed to retrieve extended switch types from the database");
```

Comment:
```suggestion
        .await?;
```

Errors can happen even if the request is correct. The API use must be notified in this case. We keep panics around only for cases that *can never* fail such as `vec.first().unwrap()` after checking that `!vec.is_empty()`.